File: LJV-7000 Series - Prolifometer/LJV_communication_library1_4_2/LJV_communication_library1_4_2/Sample/CS/Source/Sample/Forms/OutSelectForm.cs
//----------------------------------------------------------------------------- 
// <copyright file="OutSelectForm" company="KEYENCE">
//	 Copyright (c) 2013 KEYENCE CORPORATION.  All rights reserved.
// </copyright>
//----------------------------------------------------------------------------- 
using System;
using System.ComponentModel;
using System.Windows.Forms;

namespace LJV7_DllSampleAll.Forms
{
	public partial class OutSelectForm : Form
	{
		#region Field
		/// <summary>
		/// Auto zero on request and off request
		/// </summary>
		private byte _onOff;

		/// <summary>
		/// The bits of the parameter are used to specify the processing target OUTs.
		/// </summary>
		private uint _out;
		#endregion

		#region Property
		/// <summary>
		/// Other than 0: auto zero on request, 0: auto zero off request
		/// </summary>
		public byte OnOff
		{
			get { return _onOff; }
		}

		/// <summary>
		/// The bits of the parameter are used to specify the processing target OUTs.
		/// </summary>
		public uint Out
		{
			get { return _out; }
		}
		#endregion

		#region Event
		/// <summary>
		/// Close start event
		/// </summary>
		/// <param name="e"></param>
		protected override void OnClosing(CancelEventArgs e)
		{
			if (DialogResult == DialogResult.OK)
			{
				try
				{
					_onOff = Convert.ToByte(_txtboxOnOff.Text);
					_out += _chkboxBit1.Checked ? (uint)1 : (uint)0;
					_out += _chkboxBit2.Checked ? (uint)(1 << 1) : (uint)0;
					_out += _chkboxBit3.Checked ? (uint)(1 << 2) : (uint)0;
					_out += _chkboxBit4.Checked ? (uint)(1 << 3) : (uint)0;
					_out += _chkboxBit5.Checked ? (uint)(1 << 4) : (uint)0;
					_out += _chkboxBit6.Checked ? (uint)(1 << 5) : (uint)0;
					_out += _chkboxBit7.Checked ? (uint)(1 << 6) : (uint)0;
					_out += _chkboxBit8.Checked ? (uint)(1 << 7) : (uint)0;
					_out += _chkboxBit9.Checked ? (uint)(1 << 8) : (uint)0;
					_out += _chkboxBit10.Checked ? (uint)(1 << 9) : (uint)0;
					_out += _chkboxBit11.Checked ? (uint)(1 << 10) : (uint)0;
					_out += _chkboxBit12.Checked ? (uint)(1 << 11) : (uint)0;
					_out += _chkboxBit13.Checked ? (uint)(1 << 12) : (uint)0;
					_out += _chkboxBit14.Checked ? (uint)(1 << 13) : (uint)0;
					_out += _chkboxBit15.Checked ? (uint)(1 << 14) : (uint)0;
					_out += _chkboxBit16.Checked ? (uint)(1 << 15) : (uint)0;
				}
				catch (Exception ex)
				{
					MessageBox.Show(this, ex.Message);
					e.Cancel = true;
					return;
				}
			}

			base.OnClosing(e);
		}
		#endregion

		#region Method
		/// <summary>
		/// Constructor
		/// </summary>
		public OutSelectForm(bool isOnOffVisible)
		{
			InitializeComponent();

			// Field initialization
			_onOff = 0;
			_out = 0;

			// Control display control
			_lblOnOff.Visible = isOnOffVisible;
			_txtboxOnOff.Visible = isOnOffVisible;
			_lblOnOffDescription.Visible = isOnOffVisible;
		}
		#endregion
	}
}

File: LJV-7000 Series - Prolifometer/LJV_communication_library1_4_2/LJV_communication_library1_4_2/Sample/VBNET/Source/Forms/OutSelectForm.vb
'----------------------------------------------------------------------------- 
' <copyright file="OutSelectForm" company="KEYENCE">
'	 Copyright (c) 2013 KEYENCE CORPORATION.  All rights reserved.
' </copyright>
'----------------------------------------------------------------------------- 
Imports System.ComponentModel
Imports System.Windows.Forms

Namespace Forms
	Public Partial Class OutSelectForm
		Inherits Form
		#Region "Field"
		''' <summary>
		''' Auto zero on request and off request
		''' </summary>
		Private _onOff As Byte

		''' <summary>
		''' The bits of the parameter are used to specify the processing target OUTs.
		''' </summary>
		Private _out As UInteger
		#End Region

		#Region "Property"
		''' <summary>
		''' Other than 0: auto zero on request, 0: auto zero off request
		''' </summary>
		Public ReadOnly Property OnOff() As Byte
			Get
				Return _onOff
			End Get
		End Property

		''' <summary>
		''' The bits of the parameter are used to specify the processing target OUTs.
		''' </summary>
		Public ReadOnly Property Out() As UInteger
			Get
				Return _out
			End Get
		End Property
		#End Region

		#Region "Event"
		''' <summary>
		''' Close start event
		''' </summary>
		''' <param name="e"></param>
		Protected Overrides Sub OnClosing(e As CancelEventArgs)
			If DialogResult = DialogResult.OK Then
				Try
					_onOff = Convert.ToByte(_txtboxOnOff.Text)
					_out += If(_chkboxBit1.Checked, CUInt(1), CUInt(0))
					_out += If(_chkboxBit2.Checked, CUInt(1 << 1), CUInt(0))
					_out += If(_chkboxBit3.Checked, CUInt(1 << 2), CUInt(0))
					_out += If(_chkboxBit4.Checked, CUInt(1 << 3), CUInt(0))
					_out += If(_chkboxBit5.Checked, CUInt(1 << 4), CUInt(0))
					_out += If(_chkboxBit6.Checked, CUInt(1 << 5), CUInt(0))
					_out += If(_chkboxBit7.Checked, CUInt(1 << 6), CUInt(0))
					_out += If(_chkboxBit8.Checked, CUInt(1 << 7), CUInt(0))
					_out += If(_chkboxBit9.Checked, CUInt(1 << 8), CUInt(0))
					_out += If(_chkboxBit10.Checked, CUInt(1 << 9), CUInt(0))
					_out += If(_chkboxBit11.Checked, CUInt(1 << 10), CUInt(0))
					_out += If(_chkboxBit12.Checked, CUInt(1 << 11), CUInt(0))
					_out += If(_chkboxBit13.Checked, CUInt(1 << 12), CUInt(0))
					_out += If(_chkboxBit14.Checked, CUInt(1 << 13), CUInt(0))
					_out += If(_chkboxBit15.Checked, CUInt(1 << 14), CUInt(0))
					_out += If(_chkboxBit16.Checked, CUInt(1 << 15), CUInt(0))
				Catch ex As Exception
					MessageBox.Show(Me, ex.Message)
					e.Cancel = True
					Return
				End Try
			End If

			MyBase.OnClosing(e)
		End Sub
		#End Region

		#Region "Method"
		''' <summary>
		''' Constructor
		''' </summary>
		Public Sub New(isOnOffVisible As Boolean)
			InitializeComponent()

			' Field initialization
			_onOff = 0
			_out = 0

			' Control display control
			_lblOnOff.Visible = isOnOffVisible
			_txtboxOnOff.Visible = isOnOffVisible
			_lblOnOffDescription.Visible = isOnOffVisible
		End Sub
		#End Region
	End Class
End Namespace
